modules/_labs/select/react/lib/SelectMenu.tsx

```diff
@@ -1,18 +1,47 @@
-import * as React from 'react';
+import React, {useState, useEffect} from 'react';
+import {Rect} from '@popperjs/core';
+
 import {CSSObject, keyframes} from '@emotion/core';
 import {EmotionCanvasTheme, ErrorType, Themeable, styled} from '@workday/canvas-kit-react-common';
-import {colors, borderRadius, inputColors, spacingNumbers} from '@workday/canvas-kit-react-core';
+import {Placement, Popper, useCloseOnEscape} from '@workday/canvas-kit-react-popup';
+import {colors, borderRadius, inputColors} from '@workday/canvas-kit-react-core';
+
 import {SelectProps} from './Select';
+import {buttonBorderWidth} from './SelectBase';
 
 interface SelectMenuProps
   extends Themeable,
     React.HTMLAttributes<HTMLUListElement>,
     Pick<SelectProps, 'error'> {
+  /**
+   * The ref to the anchor button element. Required to anchor the menu.
+   */
+  buttonRef: React.RefObject<HTMLButtonElement>;
   /**
    * If true, enable animation on the SelectMenu.
    * @default true
    */
   isAnimated: boolean;
+  /**
+   * If true, automatically flip the SelectMenu to keep it visible if necessary (e.g., if the the SelectMenu would otherwise display below the visible area of the viewport).
+   * @default true
+   */
+  isAutoFlipped: boolean;
+  /**
+   * If true, focus the SelectMenu when it's shown. Set to false if you don't want to focus the SelectMenu automatically (for visual testing purposes, for example).
+   * @default true
+   */
+  isAutoFocused: boolean;
```

Comment:
Same with these. The intent is more like a `shouldAutoFocus` - a configuration rather than a changing state.

Reply:
Renamed `isAutoFocused` to `shouldAutoFocus` (as well as `isAnimated` to `shouldAnimate`, and `isAutoFlipped` to `shouldAutoFlip`) to better convey intent.